crates/services/upgradable-executor/wasm-executor/src/new_tx_waiter.rs

```diff
@@ -0,0 +1,12 @@
+use fuel_core_executor::{
+    executor::WaitNewTransactionsResult,
+    ports::NewTxWaiterPort,
+};
+
+pub struct NewTxWaiter;
+
+impl NewTxWaiterPort for NewTxWaiter {
```

Comment:
What's the difference between  this and `TimeoutOnlyTxWaiter`?

I don't think I mind the redundancy, just figured I'd check that it was intentional.

Reply:
The same reason: https://github.com/FuelLabs/fuel-core/pull/2799#discussion_r1982152904